portability-gateway/src/main/java/org/dataportabilityproject/gateway/reference/ReferenceApiModule.java

```diff
@@ -0,0 +1,97 @@
+/*
+* Copyright 2018 The Data-Portability Project Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.dataportabilityproject.gateway.reference;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.multibindings.MapBinder;
+import com.sun.net.httpserver.HttpHandler;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import javax.inject.Named;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Bindings the reference api server.
```

Comment:
why is this called 'reference'? not immediately clear to me

Reply:
A reference application indicates a specific implementation that developers can refer to but is not necessarily meant to serve in a production capacity. It highlights the use of APIs and functionality for those who want to implement their own version.

Reply:
oh I see. Kind of a sample implementation? Can you document it some more?

Reply:
Done